dcos_test_utils/gce.py

```diff
@@ -0,0 +1,205 @@
+
+    @retry(wait_fixed=2000, retry_on_result=_check_status)
+    def wait(self):
+        response = self.get_info()
+        log.debug(response)
+        return response
+
+    def get_instances_info(self):
+        response = self.gce_wrapper.list_deployment_instances(self)
+        log.debug(response)
+        return response
+
+    def allow_external_access(self):
+        body = {
+            "kind": "compute#firewall",
+            "name": "allow all ports",
+            "description": "allow all ports",
+            "network": "global/networks/default",
+            "sourceRanges": [
+                "0.0.0.0/0"
+            ],
+            "allowed": [
+                {
+                    "IPProtocol": "tcp",
+                    "ports": [
+                        "0-65535"
+                    ]
+                },
+                {
+                    "IPProtocol": "udp",
+                    "ports": [
+                        "0-65535"
+                    ]
+                },
+                {
+                    "IPProtocol": "icmp"
+                }
+            ]
+        }
+
+        response = self.gce_wrapper.add_firewall_rule(body)
+        log.debug(response)
+        return response
+
+
+def catch_http_exceptions(f):
+    '''
+    Runs the function "f" and raises an exception with a custom help message if a google HttpError with status 404 or
+        409 is caught
+    :param f: function to be executed
+    :return: return value of function "f"
+    '''
+    def handle_exception(*args, **kwargs):
+        try:
+            return f(*args, **kwargs)
+        except HttpError as e:
+            if e.resp.status == 404:
+                log.exception("The resource you are trying to access doesn't exist")
+            elif e.resp.status == 409:
+                log.exception('''The specified resources exist and might be under an active operation
+                                   (operation conflict)''')
+            raise e
+
+    return handle_exception
+
+
+class GceWrapper:
+    @catch_http_exceptions
+    def __init__(self, credentials_dict):
+        credentials = GoogleCredentials.get_application_default()
+        self.compute = discovery.build('compute', 'v1', credentials=credentials)
+        self.deployment_manager = discovery.build('deploymentmanager', 'v2', credentials=credentials)
+        self.project_id = credentials_dict['project_id']
+
+    @catch_http_exceptions
+    def deploy_instances(self, config):
+        body = {
+            'name': config['deployment_name'],
+            'target': {
+                'config': {
+                    'content': config['template_body']
+                }
+            }
+        }
+
+        return self.deployment_manager.deployments().insert(project=self.project_id, body=body).execute()
+
+    @catch_http_exceptions
+    def list_deployment_instances(self, deployment: Deployment):
```

Comment:
Turn this into a generator. you just need to yield instead of appending the instance